// @pnp/graph/shares/types.js
import { __decorate } from "tslib";
import { defaultPath } from "../decorators.js";
import { graphInvokableFactory, _GraphQueryableCollection, _GraphQueryableInstance } from "../graphqueryable.js";
import { DriveItem } from "../onedrive/types.js";
/**
 * Shares
 */
let _Shares = class _Shares extends _GraphQueryableCollection {
    /**
     * Gets a share by share id or encoded url
     * @param id The share id
     * @returns An IShare instance
     */
    getById(id) {
        return Share(this, id);
    }
    /**
     * Creates a sharing link (id) from a given absolute url to a file
     * @param url Absolute file url such as "https://{tenant}.sharepoint.com/sites/dev/Shared%20Documents/new.pptx"
     * @returns An encoded sharing id which can be used in getById to access a file
     */
    encodeSharingLink(url) {
        return (`u!${Buffer.from(url).toString("base64").replace(/=$/, "").replace("/", "_").replace("+", "-")}`);
    }
};
_Shares = __decorate([
    defaultPath("shares")
], _Shares);
export { _Shares };
export const Shares = graphInvokableFactory(_Shares);
/**
 * Share
 */
export class _Share extends _GraphQueryableInstance {
    /**
     * Access the driveItem associated with this shared file
     */
    get driveItem() {
        return DriveItem(this, "driveitem");
    }
}
export const Share = graphInvokableFactory(_Share);
